rust/typeql.rs

```diff
@@ -10,6 +10,8 @@
 #![deny(elided_lifetimes_in_paths)]
 #![deny(unused_must_use)]
 
+#![allow(edition_2024_expr_fragment_specifier)]
```

Comment:
`:expr` in macros will accept `const` blocks and `_` expressions in the 2024 edition. We don't currently depend on any specific behaviour in our macros.

(https://doc.rust-lang.org/nightly/edition-guide/rust-2024/macro-fragment-specifiers.html#migration)